peas/weather_web_data.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+
+import requests
+import logging
+
+import astropy.units as u
+from astropy.units import cds
+from astropy.table import Table
+from astropy.time import Time, TimeISO, TimeDelta
+
+from pocs.utils.messaging import PanMessaging
+from . import load_config
+from .weather_abstract import WeatherAbstract
+from .weather_abstract import get_mongodb
+
+
+class MixedUpTime(TimeISO):
+
+    """
+    Subclass the astropy.time.TimeISO time format to handle the mixed up
+    American style time format that the AAT met system uses.
+    """
+
+    name= 'mixed_up_time'
+    subfmts = (('date_hms',
+                '%m-%d-%Y %H:%M:%S',
+                '{mon:02d}-{day:02d}-{year:d} {hour:02d}:{min:02d}:{sec:02d}'),
+               ('date_hm',
+                '%m-%d-%Y %H:%M',
+                '{mon:02d}-{day:02d}-{year:d} {hour:02d}:{min:02d}'),
+               ('date',
+                '%m-%d-%Y',
+                '{mon:02d}-{day:02d}-{year:d}'))
+
+# -----------------------------------------------------------------------------
+# External weather data class
+# -----------------------------------------------------------------------------
+class WeatherData(WeatherAbstract):
+
+    """ Gets AAT weather data from  http://site.aao.gov.au/AATdatabase/met.html
+
+    Turns the weather data into a useable and meaningful table whose columns are the
+    entries of the data. The table only features one row (Excluding the title row)
+    which has the numerical data of all the entries.
+
+    The data is then compared with specified parameters and if the data is not within
+    the parameters then the safety condition of that entry is defined as False, i.e.
+    not safe.
+    Data is also given the weather condition, for example if the wind value is greater
+    that the one specified the weather condition will be 'windy'.
+
+    Once all the required data has been defined and given conditions, it will then
+    decide if the system is safe. If one value is False then the safety condition of
+    the system is False. All entries must be True so that the safety condition can
+    return True.
+
+    The final conditions and values are then sent to the dome controller to either
+    leave, open or close the dome. They are also saved in a database so that previous
+    entries can be retrived.
+    """
+
+    def __init__(self, use_mongo=True):
+        super().__init__(use_mongo=use_mongo)
+
+        # Read configuration
+        self.web_data = self.config['weather']['web_service']

```

Comment:
I think `self.web_data` is a misleading name for this.  It's not data, for one thing, it's configuration info, specifically the configuration info for the met data web service.  `self.web_config`?

Reply:
Yep, I agree. Does seem kind of misleading, I have changed it to `self.web_config`. On this note, is the file name `weather_web_data.py` OK? I feel like I can name it something better but I can't think of it!

Reply:
A pendant might point out that since the best source of AAT met data is an ftp site and not an http site then it is not in fact 'web' data... (though it is internet data). I wouldn't worry about the file name at this point, though.

Reply:
I have just changed it to `internet_weather.py`, just to avoid confusion.